discovery-provider/integration_tests/tasks/test_index_operations.py

```diff
@@ -197,25 +197,143 @@ def test_index_operations(celery_app, celery_app_contracts):
         # Make sure the data we added is there
         users = session.query(User).filter(User.handle == new_user_handle).all()
         tracks = session.query(Track).filter(Track.owner_id == new_user_id).all()
-
         assert len(users) > 0
         assert len(tracks) > 0
 
 
-def test_index_operations_indexing_error(celery_app, celery_app_contracts, monkeypatch):
+def test_index_operations_metadata_fetch_error(
+    celery_app, celery_app_contracts, mocker
+):
+    """
+    Confirm indexer throws IndexingError when ipfs metadata fetch throws an error
+    """
+    task = celery_app.celery.tasks["update_discovery_provider"]
+    db = task.db
+    web3 = celery_app_contracts["web3"]
+
+    # patch ipfs metadata event to raise an exception
+    def fetch_metadata_stub(*_):
+        raise Exception("Broken fetch")
+
+    mocker.patch(
+        "src.utils.ipfs_lib.IPFSClient.get_metadata",
+        side_effect=fetch_metadata_stub,
+        autospec=True,
+    )
+
+    seed_contract_data(task, celery_app_contracts, web3)
+
+    current_block = None
+    latest_block = None
+    try:
+        with db.scoped_session() as session:
```

Comment:
would be nice to have this in a helper 

def run_indexing_until_caught_up() or something

not requirement, just probs nice